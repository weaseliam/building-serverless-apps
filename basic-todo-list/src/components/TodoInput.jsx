import React, { Component } from 'react';
import PropTypes from 'prop-types';

class TodoInput extends Component {
  static propTypes = {
    onAddTodo: PropTypes.func
  }

  static defaultProps = {
    onAddTodo: () => { }
  }

  constructor() {
    super();

    this.clearValue();
  }

  clearValue = () => {
    this.state = {
      value: ''
    };
  }

  handleSubmit = (e) => {
    e.preventDefault();
    if (this.state.value === '') {
      return;
    }

    console.log(`Submit handler called with ${this.state.value}`);
    this.props.onAddTodo(this.state.value);
    this.clearValue();
  }

  handleChange = (e) => {
    this.setState({
      value: e.target.value
    });
  }

  render() {
    let { value } = this.state;

    return (
      <form onSubmit={this.handleSubmit}>
        <input type="text" onChange={this.handleChange} value={value}></input>
        <button type="submit">Add</button>
      </form>
    );
  }
};

export default TodoInput;
