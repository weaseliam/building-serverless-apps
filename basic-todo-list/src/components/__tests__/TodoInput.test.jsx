import React from 'react';
import renderer from 'react-test-renderer';

import TodoInput from '../TodoInput';

test('TodoInput renders as expected', () => {
  const component = renderer.create(<TodoInput />);

  expect(component.toJSON()).toMatchSnapshot();
});
