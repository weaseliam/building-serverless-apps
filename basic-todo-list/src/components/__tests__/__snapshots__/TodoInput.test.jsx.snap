// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoInput renders as expected 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    type="submit"
  >
    Add
  </button>
</form>
`;
